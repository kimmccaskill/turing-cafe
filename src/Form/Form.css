.reservation-form {
  width: 100%;
}

input {
  margin: 2%;
  height: 25px;
  width: 200px;
  font-size: 1em;
}

button {
  margin: 2%;
  height: 30px;
  width: 200px;
  font-size: 1em;
  color: #fff;
  background-color: teal;
}