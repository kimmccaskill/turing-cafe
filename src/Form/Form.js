import React, { Component } from 'react';
import './Form.css';

class Form extends Component {
  constructor(props) {
    super(props);
    this.state = {
      id: Date.now(),
      name: '',
      date: '',
      time: '',
      number: '',
    }
  }

  updateChange = (e) => {
    this.setState({[e.target.name]: e.target.value})
  }

  submitReservation = (e) => {
    e.preventDefault()
    this.props.addRes(this.state)
    this.setState({
      id: Date.now(),
      name: '',
      date: '',
      time: '',
      number: '',
    })
  }

  render() {
    return (
      <form
        className="reservation-form"
        onSubmit={this.submitReservation}
      >
        <input
          name='name'
          placeholder="Name"
          type="text"
          value={this.state.name}
          onChange={this.updateChange}
        />
        <input
          name='date'
          placeholder="Date (mm/dd)"
          type="text"
          value={this.state.date}
          onChange={this.updateChange}
        />
        <input
          name='time'
          placeholder="Time"
          type="text"
          value={this.state.time}
          onChange={this.updateChange}
        />
        <input
          name='number'
          placeholder="Number of guests"
          type="text"
          value={this.state.number}
          onChange={this.updateChange}
          maxlength='2'
          min='1'
          max='25'
        />
        <button 
          onClick={this.submitReservation}
          >Make a Reservation
        </button>
      </form>
    )
  }
}

export default Form;