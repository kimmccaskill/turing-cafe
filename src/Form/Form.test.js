import React from 'react';
import Form from './Form';
import { shallow } from 'enzyme';

describe('Form', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Form />)
  })

  it('should be an instance of Form component', () => {
    expect(wrapper).toMatchSnapshot();
  });
})