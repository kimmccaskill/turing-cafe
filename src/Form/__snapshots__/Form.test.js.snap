// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should be an instance of Form component 1`] = `
<form
  className="reservation-form"
  onSubmit={[Function]}
>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="Date (mm/dd)"
    type="text"
    value=""
  />
  <input
    name="time"
    onChange={[Function]}
    placeholder="Time"
    type="text"
    value=""
  />
  <input
    max="25"
    maxlength="2"
    min="1"
    name="number"
    onChange={[Function]}
    placeholder="Number of guests"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Make a Reservation
  </button>
</form>
`;
